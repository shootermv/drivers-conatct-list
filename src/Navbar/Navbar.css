header.top-navbar {
    background: var(--main-primary-color);
    display: flex;
    justify-content: space-between;
    padding: 10px 20px;
    box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
    position: sticky;
    top: 0;
}
  
header.top-navbar a{
    color: var(--main-secondary-color);
}
  
header.top-navbar form {
    display: flex;
    border-radius: 10px;
    background: #fff;
    padding: 0 10px;
}

header.top-navbar input {
    border: none;
    padding: 4px;
    outline: none;
    width: 200px;
}