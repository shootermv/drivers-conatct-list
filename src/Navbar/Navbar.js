import React from "react";
import { useSelector, useDispatch } from "react-redux";
import {filterContacts, filterCleaned} from '../store/actions';

/* ICONS */
import searchIcon from './searchIcon.svg';
import logo from './logo.svg';
import cleanBtn from './cleanBtn.svg';


import {Input, Form, Brand, Header} from  './Navbar.styles.js';


const Navbar = () => {
  const term = useSelector(({contact: {term}}) => term || '');
  const chang = e => dispatch(filterContacts(e.target.value));
  const dispatch = useDispatch();
  return (
    <Header>
        <Brand>
            <a href="/">
                <span>Contact List</span>
                <img src={logo} alt={'company logo'}/>
            </a>
        </Brand>
        <Form>
            <Input placeholder="search" value={term} onChange={chang}/>
            {term ? <img alt={'clean'} src={cleanBtn} onClick={e => dispatch(filterCleaned())}/>: <img  alt={'search'} src={searchIcon}/>}
        </Form>
    </Header>
   );
};

export default Navbar;
