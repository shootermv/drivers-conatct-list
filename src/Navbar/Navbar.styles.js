import styled from 'styled-components';


const Header = styled.header`
    background: var(--main-primary-color);
    display: flex;
    justify-content: space-between;
    padding: 10px 20px;
    box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
    position: sticky;
    top: 0;
    z-index:2;
`
  
const Brand = styled.div`
   a span{display:none;}
   a img {
       height: 30px;
       margin: 5px 10px 0  0; 
   } 
   /* mobile styles */

   /* small tablet styles */
   @media screen and (min-width: 620px){
      a img{display:none;} 
      a span{
          display:block;
          color: var(--main-secondary-color);
      }
   }
`


const Form = styled.form`
    display: flex;
    border-radius: 10px;
    background: var(--main-secondary-color);
    padding: 0 10px;
    width: 100%;
    img {
        height: 20px;
        margin: 10px -5px 0 0;
        opacity: 0.2;
    }
    /* mobile styles */

    /* small tablet styles */
    @media screen and (min-width: 620px){
        &{
          width: 200px;
        }
        img {
            height: 15px;
            margin: 3px -5px 0 0;
            opacity: 0.2;
        }
    }      
`

const Input = styled.input`
    border: none;
    padding: 4px;
    outline: none;
    width: 100%;
    ::placeholder {
        opacity: 0.8
    }
  
`

export {Input, Form, Brand, Header}