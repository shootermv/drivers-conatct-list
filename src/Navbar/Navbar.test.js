import React from 'react'
import { render, fireEvent } from '@testing-library/react';

import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from '../store/reducers';
import Navbar from './Navbar'

describe('Navbar spec', () => {
    const mockStore = createStore(reducer, {term: null, });

    it('renders the component', () => {
      const container = render(
        <Provider store={mockStore}>
           <Navbar/>
        </Provider>
      )
      expect(container.firstChild).toMatchSnapshot()
     })
})    