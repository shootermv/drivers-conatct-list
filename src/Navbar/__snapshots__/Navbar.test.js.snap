// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar spec renders the component 1`] = `undefined`;
